plugins/config_manager.py

```diff
@@ -0,0 +1,241 @@
+                dnfpluginscore.lib.write_raw_configfile(repo.repofile,
+                                                        repo.id,
+                                                        sbc.substitutions,
+                                                        repo.cfg.options,
+                                                        repo.iteritems,
+                                                        repo.optionobj,
+                                                        repo_modify)
+
+    def add_repo(self):
+        """ process --add-repo option """
+
+        # put repo file into first reposdir which exists or create it
+        myrepodir = None
+        for rdir in self.base.conf.reposdir:
+            if os.path.exists(rdir):
+                myrepodir = rdir
+                break
+
+        if not myrepodir:
+            myrepodir = self.base.conf.reposdir[0]
+            dnf.util.ensure_dir(myrepodir)
+
+        for url in self.opts.add_repo:
+            if urlparse.urlparse(url).scheme == '':
+                url = 'file://' + os.path.abspath(url)
+            logger.info(_('Adding repo from: %s'), url)
+            if url.endswith('.repo'):
+                # .repo file - download, put into reposdir and enable it
+                destname = os.path.basename(url)
+                destname = myrepodir + '/' + destname
+                try:
+                    with dnfpluginscore.lib.urlopen(self, None, url) as f:
+                        if not save_to_file(destname, f.read()):
+                            continue
+                except IOError as e:
+                    logger.error(e)
+                    continue
+            else:
+                # just url to repo, create .repo file on our own
+                repoid = sanitize_url_to_fs(url)
+                reponame = 'created by dnf config-manager from %s' % url
+                destname = myrepodir + '/' + repoid + '.repo'
+                content = "[%s]\nname=%s\nbaseurl=%s\nenabled=1\n" % \
+                                                (repoid, reponame, url)
+                if not save_to_file(destname, content):
+                    continue
+
+
+def save_to_file(filename, content):
+    try:
+        with open(filename, 'w+') as fd:
+            fd.write(content)
+    except (IOError, OSError) as e:
+        logger.error(_('Could not save repo to repofile %s: %s'),
+                     filename, e)
+        return False
+    return True
+
+# Regular expressions to sanitise cache filenames
+RE_SCHEME = re.compile(r'^\w+:/*(\w+:|www\.)?')
+RE_SLASH = re.compile(r'[?/:&#|]+')
+RE_BEGIN = re.compile(r'^[,.]*')
+RE_FINAL = re.compile(r'[,.]*$')
+
+def sanitize_url_to_fs(url):
+    """Return a filename suitable for the filesystem
+
+    Strips dangerous and common characters to create a filename we
+    can use to store the cache in.
+    """
+
+    # code taken and modified from planet venus code base:
+    # http://intertwingly.net/code/venus/LICENCE
```

Comment:
what does this mean to the license of the DNF package? no change needed (even in the spec file)?